egui/src/widgets/text_edit.rs

```diff
@@ -656,7 +665,18 @@ impl<'t> TextEdit<'t> {
 
         ui.memory().id_data.insert(id, state);
 
-        response.widget_info(|| WidgetInfo::text_edit(&*text));
+        if response.changed {
+            response.widget_info(|| WidgetInfo::text_edit(&*text, &*prev_text));
+        } else if let Some(text_cursor) = text_cursor {
+            response.has_widget_info = true;
+            response.widget_info(|| {
+                WidgetInfo::text_selection_changed(
```

Comment:
This emits text selection change every frame, no? That's a bit excessive!

We should be able to detect when the cursor changes and only emit it then

Reply:
You're right, it does. Have to admit to being a bit overwhelmed with all of this. :) How do I get the previous cursor location? Right now I'm stashing that state locally in the screen reader and checking it for changes, but obviously if I can kick that can back down the road to egui then I'm happy to do it.

Reply:
I see `output.text_cursor`. Is that only the cursor position? I'd need start and end values to determine if the value changed.

Reply:
The simplest is probably to store the text cursor at the start of `TextEdit::content_ui` and then compare against that here.